Sources/TensorFlow/Loss.swift

```diff
@@ -290,3 +290,31 @@ public func sigmoidCrossEntropy<Scalar: TensorFlowFloatingPoint>(
     let negAbsLogits = max(logits, -logits) // Custom `abs` to compute gradients at `0`.
     return reduction(maxLogitsWithZero - logits * labels + log1p(exp(-negAbsLogits)))
 }
+
+/// Returns the Huber loss between predictions and expectations.
+///
+/// For each value x in error = exprected - predicted
+///     loss = 0.5 * x^2                 if |x| <= ∂
+///     loss = 0.5 * ∂^2 + ∂ * (|x| - ∂) if |x| > ∂
+/// See: https://en.wikipedia.org/wiki/Huber_loss
+///
+/// - Parameters:
+///   - predicted: Predicted outputs from a neural network.
+///   - expected: Expected values, i.e. targets, that correspond to the correct output.
+///   - delta: A floating point scalar, the point where the Huber loss function changes from a
+///     quadratic to linear. 
+///   - reduction: Reduction to apply on the computed element-wise loss values.
+@differentiable(wrt: predicted)
+public func huberLoss<Scalar: TensorFlowFloatingPoint>(
+    predicted: Tensor<Scalar>,
+    expected: Tensor<Scalar>,
+    delta: Scalar,
+    reduction: @differentiable (Tensor<Scalar>) -> Tensor<Scalar> = { $0.sum() }
```

Comment:
Just a note: some other `reduction: @differentiable (Tensor<Scalar>) -> Tensor<Scalar` default arguments are using a special function `_mean` due to [some linker errors](https://bugs.swift.org/projects/TF/issues/TF-690).

I'm not sure if these linker errors are still reproducible, but the underlying infrastructure regarding these linker issues will be revamped soon.

Reply:
@dan-zheng Do you suggest to use _mean for the moment and eventually change it later on ?I'm able to run the test on my computer with the toolchain 0.5 but I don't know if it means something.